62/Gealber/scrapper.py

```diff
@@ -0,0 +1,140 @@
+                speakers = speakers + speaker
+            return speakers
+
+        elif year in years_common:
+            items = soup.select("div.slot-inner > p > strong")
+            for i, link in enumerate(items):
+                s = str(link)
+                start_content = s.find('<strong>')
+                end_content = s.find('</strong>', start_content + 1)
+                speaker = str(s[start_content + 8: end_content]).replace(' y', ',').split(', ')
+                speakers = speakers + speaker
+            return speakers
+        elif year == 2014:
+            items = soup.select("h1.text-center")
+            for i, link in enumerate(items):
+                s = str(link)
+                start_content = s.find('ter">')
+                end_content = s.find('</h1>', start_content + 1)
+                speaker = str(s[start_content + 5: end_content]).replace(' y', ',').split(', ')
+                speakers = speakers + speaker
+            return speakers
+        elif year == 2013:
+            items = soup.select('li > div > a[target="_blank"]')
+            for i, link in enumerate(items):
+                s = str(link)
+                start_content = s.find('">')
+                end_content = s.find('</a>', start_content + 1)
+                speaker = str(s[start_content + 2: end_content]).replace(' y', ',').split(', ')
+                speakers = speakers + speaker
+            return speakers
+
+    def do_classify_in_threads(self) -> dict:
+        """
+        Open the the URLS and scrape the info related to the speakers.
+        After that store the info in a dictionary to be classify by gender.
+        Return a dictionary with the following form {'2016': [1, 0, 0, 1], '2018': [0, 1, 0]}
+        where 1 represent a female and 0 a male.
+        :return:
+        """
+        speakers_in_year_es = {}
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=6) as executor:
+            future_to_url = {executor.submit(self.connect, url): url for url in URLS_ES}
+            for future in concurrent.futures.as_completed(future_to_url):
+                url = future_to_url[future]
+                try:
+                    page = future.result()[0]
+                    year = future.result()[1]
+                    speakers_in_year_es[str(year)] = self.scrapper_es(page, year)
+                except Exception as exc:
+                    print(f"{url} generated the following exception: \n{exc}")
+
+        # With the next loops we get a dictionary
+        # of this form {'2016': ['female', 'male', 'female']}
+        # and then we replace 'female' by 1 and other cases by 0.
+        d = gender.Detector()
+        for year in speakers_in_year_es:
+            for i, name in enumerate(speakers_in_year_es[year]):
+                speakers_in_year_es[year][i] = 1 if d.get_gender(name.split()[0]) == 'female' else 0
```

Comment:
This is not very Pythonic and could be rewritten using list comprehensions, at least for the inner loop. No need for the "i"